cmd/validate.go

```diff
@@ -289,7 +297,11 @@ func processEntry(ctx context.Context, pw progress.Writer, wp cmdutil.Pool, e pl
 					continue
 				}
 
-				ccrit := newCriteria(entry)
+				ccrit, err := newCriteria(entry)
```

Comment:
We don't need to shell out here, so I'd prefer to separate criteria as a hash object from how we stringify it (which needs more info from the schema).

Reply:
plugin.Schema is going to shell out if the external plugin doesn't prefetch its schemas, so not sure what you mean here.

Reply:
Didn't we separate TypeID so you don't need to call `schema` for it? That's the only one required by the hash. The label and singleton are only used when deciding how to display it, which we do once at the beginning.

Reply:
Oh I see what you're saying. Will update.